src/models/admin.jsx

```diff
@@ -79,12 +80,27 @@ const AdminContextProvider = ({ children }) => {
     }
   };
 
+  const deleteUser = async (userId) => {
+    const userSaveCollection = collection(database, "Users", userId, "JobsSaved");
+    const userJobs = await getDocs(userSaveCollection);
+    const awaitIds = userJobs.docs.map(async (doc) => {
+      return doc.id;
+    });
+    const jobsToDelete = await Promise.all(awaitIds);
+    await deleteDoc(doc(database, "Users", userId));
+    const jobDeleted = jobsToDelete.map(async (jobId) => {
+      await deleteDoc(doc(database, "Jobs", jobId));
+      return null;
+    });
+    await Promise.all(jobDeleted);
+  };
+
```

Comment:
- you only get the job saved by the user. we also need to do with jobs created by the user(if any)
- use `runTransaction` to let all operations as one atomic operation

Reply:
ah I see